tests/unit/fem/test_fa_determinism.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+using namespace mfem;
+
+TEST_CASE("FA Determinism", "[PartialAssembly][CUDA]")
+{
+   const int order = 3;
+   const char *mesh_filename = "../../data/star-q3.mesh";
+
+   Mesh mesh = Mesh::LoadFromFile(mesh_filename);
+
+   const int dim = mesh.Dimension();
+
+   H1_FECollection fec(order, dim);
+   FiniteElementSpace fes(&mesh, &fec);
+
+   Vector x, b_reference;
+
+   const int ntrials = 100 + 1;
```

Comment:
Probably don't need 100 trials here. I think when I tried it, the non-deterministic one was always failing on the first trial.

Reply:
done in [03ce7c1](https://github.com/mfem/mfem/pull/3172/commits/03ce7c1d4a98f589b2ba4114b5c1a6129e926ea2).

Reply:
I think `ntrials` has to be at least 2 or it won’t perform the check (first time is setting the baseline)